tests/functional/test_install_wheel.py

```diff
@@ -575,6 +576,88 @@ def test_wheel_install_with_no_cache_dir(script, tmpdir, data):
     result.assert_installed('simpledist', editable=False)
 
 
+def test_wheel_install_with_ca_pool(script, tmpdir, data):
+    """Check wheel installations work when set to use a content addressable pool.
+    """
+    package = data.packages.joinpath("simple.dist-0.1-py2.py3-none-any.whl")
+    result = script.pip(
+        'install',
+        '--use-feature=content-addressable-pool',
+        '--content-addressable-pool-save-files',
+        '--cache-dir', './cache',
+        package,
+        allow_stderr_warning=True
+    )
+
+    # This is the sha of the simple.dist-0.1-py2.py3-none-any.whl wheel file
+    digest_name, b64hash = \
+        'sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU'.split('=', 1)
+    path = urlsafe_b64decode(f'{b64hash}=').hex()
+
+    # Make sure the path of the wheel we installed is safely installed in the cacache
+    exists = [
+        os.path.exists(
+            os.path.join(
+                script.cwd,
+                "cache/pool",
+                path[:2],
+                path[2:4],
+                path[4:6],
+                path[6:],
+            ),
+        ),
+    ]
+
+    result.assert_installed('simpledist', editable=False)
+
+    assert any(exists)
+
+
+def test_wheel_install_with_ca_pool_without_saving(script, tmpdir, data):
+    """Check wheel installations work when set to use a content addressable pool,
+    but without the save option
+    """
+    package = data.packages.joinpath("simple.dist-0.1-py2.py3-none-any.whl")
+    result = script.pip(
+        'install',
+        '--use-feature=content-addressable-pool',
+        '--cache-dir', './cache',
+        package,
+        allow_stderr_warning=True
+    )
+
+    exists = [
+        os.path.exists(script.cwd / "cache/pool"),
+    ]
+
+    result.assert_installed('simpledist', editable=False)
+
+    assert not any(exists)
+
+
+def test_wheel_install_with_ca_pool_without_cache_dir(script, tmpdir, data):
```

Comment:
can you add one last test where the cache is populated, and a later installation uses the hardlinks (same inodes) from the previous attempt? or maybe check the symlink variant for coverage.

Reply:
Gonna do both :stuck_out_tongue: